src/main/java/seedu/task/logic/commands/AddTaskCommand.java

```diff
@@ -22,13 +22,20 @@
 	 *             if any of the raw values are invalid
 	 */
 
-	public AddTaskCommand(String name, String description, String deadline) throws IllegalValueException {
-		this.toAdd = new Task(new Name(name), new Description(description), new Deadline(deadline), DEFAULT_STATUS);
+	public AddTaskCommand(String name, 
+	        boolean isDescriptionAdded, String description, 
+	        boolean isDeadlineAdded, String deadline
+	        ) throws IllegalValueException {
+	    if (isDescriptionAdded && isDeadlineAdded) {
```

Comment:
Just wondering if there's a need for so many arguments in the constructor. Why not if the "description" is an empty string, it means that there is no description to be added? Same goes for the deadline string.